utils/providers.py

```diff
@@ -647,12 +649,42 @@ def get_credentials(credential_dict, cred_type=None):
     """
     domain = credential_dict.get('domain', None)
     token = credential_dict.get('token', None)
+    service_account = credential_dict.get('service_account', None)
+    if service_account:
+        service_account = gce_service_account_formating(service_account)
     return BaseProvider.Credential(
         principal=credential_dict['username'],
         secret=credential_dict['password'],
         cred_type=cred_type,
         domain=domain,
-        token=token)
+        token=token,
+        service_account=service_account)
+
+
+def gce_service_account_formating(data):
+    service_data = '''
```

Comment:
For mgmt we use credentials in yaml format, this was done only as a workaround to paste all this data in UI text area and it should be change to file upload in near future (probably today's build, went to check . . .)